beacon-chain/rpc/statefetcher/fetcher.go

```diff
@@ -187,24 +194,28 @@ func (p *StateProvider) stateByHex(ctx context.Context, stateId []byte) (state.B
 	return nil, &stateNotFoundErr
 }
 
-func (p *StateProvider) stateBySlot(ctx context.Context, slot types.Slot) (state.BeaconState, error) {
-	currentSlot := p.GenesisTimeFetcher.CurrentSlot()
-	if slot > currentSlot {
-		return nil, errors.New("slot cannot be in the future")
-	}
-	state, err := p.StateGenService.StateBySlot(ctx, slot)
+// StateBySlot returns a state that has had process_block applied to the state if a
+// canonical block is available for the slot, otherwise process_block will have been applied
+// to all states w/ canonical blocks leading up to and including the requested slot, w/ process_slots applied
+// for all subsequent slots.
```

Comment:
I find it a bit hard to understand this.

> leading up to and including the requested slot

Shouldn't `and including`be removed because it will be covered by the first scenario (a canonical block is available for the slot)?

> w/ process_slots applied for all subsequent slots

Why would I want to process anything beyond the target slot?

Reply:
updated with an attempt at a more clear comment, does this track better?
```
// StateBySlot returns the post-state for the requested slot. To generate the state, it uses the
// most recent canonical state prior to the target slot, and all canonical blocks
// between the found state's slot and the target slot.
// process_blocks is applied for all canonical blocks, and process_slots is called for any skipped
// slots, or slots following the most recent canonical block up to and including the target slot.
```